2.promise/index.js

```diff
@@ -3,6 +3,22 @@
 function fetchData(url) {
   // you can use fetch to get a Promise from URL
   // start coding here
+  fetch(url)
+    .then(response => {
+      if (!response.ok) {
+        throw new Error(`Error: ${response.statusText}`);
+      }
+      return response.json(); 
+    })
+    .then(data => {
+      let name = data.name;
+      let location = data.location || 'Not Found'; 
+      
+      console.log(`${name} lives in ${location}`);
+    })
+    .catch(error => {
+      console.error(error.message);
```

Comment:
handlingnya sudah betul, namun outputnya cukup not found saja, karena exception di sini sudah expected